### ISSUE TITLE
Autcmd "format on save" gives error with :wq

### ISSUE THREAD

**Issue by dakush** [mhartington/formatter.nvim#125]
I'm using the autocmd you suggest to format on save:
```
vim.api.nvim_exec([[
augroup FormatAutogroup
  autocmd!
  autocmd BufWritePost *.js,*.rs,*.lua FormatWrite
augroup END
]], true)
```

When I use `:wq` I always get this error when I'm back on my terminal:

`Formatter: Buffer changed while formatting, not applying formatting`

When I use just `:w` I get that same error the 1st time, but if I repeat the command then it works without complaining.

Any idea why this could be happening? Thanks!

**Comment by lukas-reineke** [comment]
You can't use `:wq`. Because formatting is done asynchronously, Neovim will already be closed by the time the formatting results come back.
This could be fixed with something like #109, but it is not currently possible.

That you get `Formatter: Buffer changed while formatting, not applying formatting` with just `:w` probably means that something else is also formatting the buffer on write.

**Comment by dakush** [comment]
OK then. Thanks for the explanation anyway!
